### PR TITLE
PLAT-128492: Add UI tests and unit tests for SliderButton

### PR DESCRIPTION
**Pull request by adrian-cocoara-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
 - Added ui tests for SliderButton
 - Added unit tests for SliderButton

### Links
[//]: # (Related issues, references)
PLAT-128492

### Comments

Enact-DCO-1.0-Signed-off-by: Adrian Cocoara adrian.cocoara@lgepartner.com

### CODE REVIEW

**Review by daniel-stoian-lgp — APPROVED** [review]

File: SliderButton/tests/SliderButton-specs.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import {mount} from 'enzyme';
+import React from 'react';
+import SliderButton from '../SliderButton';
+
+const children = ['Light Speed', 'Ridiculous Speed', 'Ludicrous Speed'];
+
+describe('SliderButton Specs', () => {
+	test('should change value when clicking on the second button', () => {
```
Comment (jeongran1230) [inline SliderButton/tests/SliderButton-specs.js:8]: I think we can make it two different test cases.
One for emitting 'onChange' event and the other is related to set 'aria-valuetext'.
IMHO, having an assertion is good in a test method if possible. 
I don't need to prove onChange event is emitted but most of unit tests in enact have it. So, If you think it is necessary, please add it. 

```suggestion
	test('should emit an onChange event when changing value', () => {z
	const handleChange = jest.fn();
		const evt = {value: 1};
		const sliderButton = mount(
			<SliderButton onChange={handleChange}>
				{children}
			</SliderButton>
		);

		sliderButton.find('SliderButton').simulate('change', evt);

		const expected = 1;
		const actual = handleChange.mock.calls[0][0].value;
		expect(actual).toBe(expected);
    });

    test('should set "aria-valuetext" to value when value is changed', () => {
		const handleChange = jest.fn();
		const evt = {value: 1};
		const sliderButton = mount(
			<SliderButton onChange={handleChange}>
				{children}
			</SliderButton>
		);

		sliderButton.find('SliderButton').simulate('change', evt);

		const valueText = sliderButton.find('SliderButton').prop('aria-valuetext');
		expect(valueText).toBe('Ridiculous Speed');
	});
```

File: tests/ui/specs/SliderButton/SliderButtonPage.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	}
+
+	get self () {
+		return $(this.selector);
+	}
+	knobValue () {
+		return $(this.selector + ' .SliderButton_SliderButton_knob');
+	}
+	clickableItem (index) {
+		return $(this.selector + ` .SliderButton_SliderButton_track>div:nth-child(${index})`);
```
Comment (jeongran1230) [inline tests/ui/specs/SliderButton/SliderButtonPage.js:17]: How about the below code for finding clickable items? I think it is more clear but you have to change then index number overall.
```suggestion
	const parent = $(this.selector + ' .SliderButton_SliderButton_track');
        return = parent.$$('.SliderButton_SliderButton_client')[index];
```

File: tests/ui/specs/SliderButton/SliderButtonPage.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+class SliderButtonInterface {
+	constructor (id) {
+		this.id = id;
+		this.selector = `#${this.id}`;
+	}
+
+	get self () {
+		return $(this.selector);
+	}
+	knobValue () {
```
Comment (jeongran1230) [inline tests/ui/specs/SliderButton/SliderButtonPage.js:13]: how about 'getKnob ' for method name?

**Comment by jeongran1230** [comment]
@adrian-cocoara-lgp, Thank you for accepting my review. I a little bit changed the code.

**Pull request merged**